erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -122,8 +122,8 @@ def get_items_with_location_and_quantity(item_doc, item_location_map):
 			'qty': qty,
 			'stock_qty': stock_qty,
 			'warehouse': item_location.warehouse,
-			'serial_no': serial_nos if auto_set_serial_no else None,
-			'batch_no': item_location.batch_no if auto_set_batch_no else None
+			'serial_no': serial_nos if auto_set_serial_no else item_doc.serial_no,
+			'batch_no': item_location.batch_no if auto_set_batch_no else item_doc.batch_no
```

Comment:
I don't like this lol

Reply:
lol, it'll be None in their case, and the batch from SO for us

win-win